idp_node/globus.py

```diff
@@ -243,22 +241,18 @@ def _install_globus(config_type):
 
 def create_globus_account(globus_sys_acct):
     '''Create the system account for globus to run as.'''
-
-    try:
-        esg_functions.call_subprocess("groupadd -r {}".format(globus_sys_acct))
-    except SubprocessError, error:
-        logger.debug("error: %s", error)
-        logger.debug("error: %s", error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
-            pass
+    esg_functions.add_unix_group(globus_sys_acct)
 
     globus_sys_acct_passwd = esg_functions.get_security_admin_password()
+
+    useradd_options = ["-s", "-r", "-c", "Globus System User", "-g", "globus", "-p", globus_sys_acct_passwd, "-s", "/bin/bash", globus_sys_acct]
     try:
-        esg_functions.stream_subprocess_output('''/usr/sbin/useradd -r -c "Globus System User" -g {globus_sys_acct_group} -p {globus_sys_acct_passwd} -s /bin/bash {globus_sys_acct}'''.format(globus_sys_acct_group="globus", globus_sys_acct_passwd=globus_sys_acct_passwd, globus_sys_acct=globus_sys_acct))
-    except SubprocessError, error:
-        logger.debug(error.__dict__["data"]["returncode"])
-        if error.__dict__["data"]["returncode"] == 9:
+        esg_functions.call_binary("useradd", useradd_options)
+    except ProcessExecutionError, err:
+        if err.retcode == 9:
             pass
+        else:
```

Comment:
I've seen this `useradd` pattern elsewhere. Could we have a `create_unix_user`, just like `create_unix_group`?

Reply:
Or simply `create_user`, `create_group`?